'SSLv3_method' vs 'SSLv23_method'

```
/src/Native/System.Security.Cryptography.Native/pal_ssl.cpp:33:32: error: use of undeclared identifier 'SSLv3_method'; did you
      mean 'SSLv23_method'?
    const SSL_METHOD* method = SSLv3_method();
                               ^~~~~~~~~~~~
                               SSLv23_method                                                                                                              
/usr/include/openssl/ssl.h:2360:19: note: 'SSLv23_method' declared here
const SSL_METHOD *SSLv23_method(void); /* Negotiate highest available SSL/TLS
```
```
$ openssl version
OpenSSL 1.0.2g  1 Mar 2016
```
```
#ifndef OPENSSL_NO_SSL3_METHOD
 const SSL_METHOD *SSLv3_method(void);
 const SSL_METHOD *SSLv3_server_method(void);
 const SSL_METHOD *SSLv3_client_method(void);
 #endif
```
https://www.openssl.org/docs/manmaster/ssl/SSL_CTX_new.html
SSLv23_* - Use of these functions is deprecated.
Ideally should be shifting to TLS now? (edit: nvm, I see it's an option to use in the enum SSLProtocols)
https://issues.asterisk.org/jira/browse/ASTERISK-24799

--- a/src/Native/System.Security.Cryptography.Native/pal_ssl.cpp
+++ b/src/Native/System.Security.Cryptography.Native/pal_ssl.cpp
@@ -30,9 +30,13 @@ extern "C" const SSL_METHOD* CryptoNative_SslV2_3Method()
 
 extern "C" const SSL_METHOD* CryptoNative_SslV3Method()
 {
+#ifdef OPENSSL_NO_SSL3_METHOD
+    return nullptr;
+#else
     const SSL_METHOD* method = SSLv3_method();
     assert(method != nullptr);
     return method;
+#endif
 }
 
 extern "C" const SSL_METHOD* CryptoNative_TlsV1Method()
@@ -86,10 +90,12 @@ extern "C" void CryptoNative_SetProtocolOptions(SSL_CTX* ctx, SslProtocols proto
     {
         protocolOptions |= SSL_OP_NO_SSLv2;
     }
+#ifndef OPENSSL_NO_SSL3
     if ((protocols & PAL_SSL_SSL3) != PAL_SSL_SSL3)
     {
         protocolOptions |= SSL_OP_NO_SSLv3;
     }
+#endif
     if ((protocols & PAL_SSL_TLS) != PAL_SSL_TLS)
     {
         protocolOptions |= SSL_OP_NO_TLSv1;

